internal/core/metadata/application/device.go

```diff
@@ -237,3 +269,40 @@ func DevicesByProfileName(offset int, limit int, profileName string, dic *di.Con
 	}
 	return devices, totalCount, nil
 }
+
+var noMessagingClientError = errors.NewCommonEdgeXWrapper(goErrors.New("unable to publish Device System Event: MessageBus Client not available"))
+
+func publishDeviceSystemEvent(action string, d models.Device, ctx context.Context, lc logger.LoggingClient, dic *di.Container) errors.EdgeX {
+	device := dtos.FromDeviceModelToDTO(d)
+	systemEvent := dtos.NewSystemEvent(common.DeviceSystemEventType, action, common.CoreMetaDataServiceKey, device.ServiceName, nil, device)
+
+	messagingClient := bootstrapContainer.MessagingClientFrom(dic.Get)
+	if messagingClient == nil {
+		return &noMessagingClientError
```

Comment:
you might need to create the error in this way:
declare the error with a normal error in line 273:
`var noMessagingClientError = goErrors.New("unable to publish Device System Event: MessageBus Client not available")`

and wrap the error for return here
`return errors.NewCommonEdgeXWrapper(noMessagingClientError)`

In this way, the errors.CommonEdgeX can record which line in which file cause this error in the debug message.
Users can still use `errors.As` to determine whether the error contains `noMessagingClientError`

Reply:
Ok, sound good.

Reply:
No loner returning error, just logging it.